### DISCUSSION TITLE
Tailwind 3, safelist and get a css classes without any markup?

### DISCUSSION THREAD

**Discussion by zoran-tietoevry** [tailwindlabs/tailwindcss#6492] [Help]
Hi,

I want to use Tailwinds grid for a tool where the user will have the possibility to make a grid based on a number of dropdown selections. I'm using Vue 3 (not with Vite). I'm not using Tailwind for any other css in the project so there will be no markup with Tw classes.

I figured I could do a "safelist" with all the classes and run a build and it would get generated but I'm having issues and not sure if this is possible.

I'm using Vue 3 but with the "Tailwind CLI" approach as described in the [documentation](https://tailwindcss.com/docs/installation).

My configuration is large so will shorten it for the sake of the question but you can find the full thing on [tailwind playground](https://play.tailwindcss.com/ZRKb7x4T4y)

```
module.exports = {
  safelist: [
    'grid-cols-1',
    'grid-cols-2',
    'grid-rows-1',
    'grid-rows-2',
    'grid-rows-3'
  ],
  prefix: 'm-',
  important: false,
  corePlugins: {
    preflight: false
  },
  theme: {},
  plugins: []
}

```

I use the following command to run it. It does generate the `tailwind.build.css` but the file is empty. 

`"build:tw": "npx tailwindcss -i ./src/assets/tailwind.css -o ./src/assets/tailwind.build.css"`

I have followed the instructions to install the tailwindcss npm package. What am I missing or is it just that simple that I can't to this? And what is requirements when using a JS file for tailwind to recognise classes?

**Comment by tobiaszciesielski** [comment]
I have the same issue but using RegExp. Final file is missing classes which are included in safelist


```
module.exports = {
  content: ['./src/**/*.{html,ts}'],

  safelist: [/col-start-\d/, /row-start-\d/, /col-span-\d/, /row-span-\d/],
  ....
}

```

Edit. Oops. My fault! I should use it this way https://tailwindcss.com/docs/content-configuration#using-regular-expressions
